toolchain/parse/BUILD

```diff
@@ -47,6 +48,35 @@ cc_test(
     ],
 )
 
+cc_library(
+    name = "parse",
+    srcs = [
+        "context.cpp",
+        "context.h",
+        "parse.cpp",
+    ] +
+    # Glob handler files to avoid missing any.
+    glob([
+        "handle_*.cpp",
+    ]),
```

Comment:
Do we still need these to be combined together? It looks like we have some nice layering we could extract now:

```mermaid
graph BT
  parse --> handlers & context & tree
  handlers --> context & tree
  context --> tree
```

(We couldn't do this before only because `Parse` was inside tree, creating a cycle.)

Reply:
I can look at that in a separate PR -- I wasn't trying to split that out here because the handlers are still defined in context.h, so I hadn't been sure of the value for splitting.